[MODULES-4326] Edit README.md to prep for localization and Puppet Strings

wilson208 opened the pull request.

pmcmaw commented on README.md:
```diff
@@ -767,169 +1206,248 @@ The description is an an optional string for a human-readable description of thi
 ##### `resource_name`
 
-The name of the Resource to be created, relative to the java:comp/env context. Default: `$name`
+The name of the Resource to be created, relative to the java:comp/env context.
```
Backticks around `java:comp/env`

pmcmaw commented on README.md:
```diff
@@ -723,35 +1144,53 @@ Specifies a Manager element in the designated xml configuration.
 ##### `additional_attributes`
 
-Specifies any additional attributes to add to the Manager. Should be a hash of the format 'attribute' => 'value'. This parameter is optional.
+Specifies any additional attributes to add to the Manager. Should be a hash of the format 'attribute' => 'value'.
```
I would put "Should be a hash of the format 'attribute' => 'value'." on a new line as there seems to be more than one of them, meaning translation would only need to happen once for the specific sentence.

pmcmaw commented on README.md:
```diff
@@ -723,35 +1144,53 @@ Specifies a Manager element in the designated xml configuration.
 ##### `attributes_to_remove`
 
-Specifies any attributes to remove from the Manager. Should be a hash of the format 'attribute' => 'value'. This parameter is optional.
+Specifies any attributes to remove from the Manager. Should be a hash of the format 'attribute' => 'value'.
```
I would put "Should be a hash of the format 'attribute' => 'value'." on a new line as there seems to be more than one of them, meaning translation would only need to happen once for the specific sentence.

pmcmaw commented on README.md:
```diff
@@ -767,169 +1206,248 @@ The description is an an optional string for a human-readable description of thi
 ##### `additional_attributes`
 
-Specifies any additional attributes to add to the Environment. Should be a hash of the format 'attribute' => 'value'. This parameter is optional.
+Specifies any additional attributes to add to the Environment. Should be a hash of the format 'attribute' => 'value'.
```
New line for Should be hash...etc

pmcmaw commented on README.md:
```diff
@@ -767,169 +1206,248 @@ The description is an an optional string for a human-readable description of thi
 ##### `attributes_to_remove`
 
-Specifies any attributes to remove from the Environment. Should be a hash of the format 'attribute' => 'value'. This parameter is optional.
+Specifies any attributes to remove from the Environment. Should be a hash of the format 'attribute' => 'value'.
```
New line for Should be hash...etc

pmcmaw commented on README.md:
```diff
@@ -767,169 +1206,248 @@ The description is an an optional string for a human-readable description of thi
 ##### `additional_attributes`
 
-Specifies any additional attributes to add to the Valve. Should be a hash of the format 'attribute' => 'value'. This parameter is optional.
+Specifies any additional attributes to add to the Valve. Should be a hash of the format 'attribute' => 'value'. 
```
New line for Should be hash...etc

pmcmaw commented on README.md:
```diff
@@ -767,169 +1206,248 @@ The description is an an optional string for a human-readable description of thi
 ##### `attributes_to_remove`
 
-Specifies any attributes to remove from the Valve. Should be a hash of the format 'attribute' => 'value'. This parameter is optional.
+Specifies any attributes to remove from the Valve. Should be a hash of the format 'attribute' => 'value'.
```
New line for Should be hash...etc

pmcmaw commented on README.md:
```diff
@@ -767,169 +1206,248 @@ The description is an an optional string for a human-readable description of thi
 ##### `additional_attributes`
 
-Specifies any additional attributes to add to the Valve. Should be a hash of the format 'attribute' => 'value'. This parameter is optional.
+Specifies any additional attributes to add to the Valve. Should be a hash of the format 'attribute' => 'value'.
```
New line for Should be hash...etc

pmcmaw commented on README.md:
```diff
@@ -767,169 +1206,248 @@ The description is an an optional string for a human-readable description of thi
 ##### `attributes_to_remove`
 
-Specifies any attributes to remove from the Valve. Should be a hash of the format 'attribute' => 'value'. This parameter is optional.
+Specifies any attributes to remove from the Valve. Should be a hash of the format 'attribute' => 'value'.
```
New line for Should be hash...etc

pmcmaw commented on README.md:
```diff
@@ -767,169 +1206,248 @@ The description is an an optional string for a human-readable description of thi
+Optional
 
 #### tomcat::config::context::resourcelink
 Specifies a ResourceLink element in the designated xml configuration.
```
Need to put this sentence on a new line.

pmcmaw commented on README.md:
```diff
@@ -767,169 +1206,248 @@ The description is an an optional string for a human-readable description of thi
 ##### `resourcelink_name`
 
-The name of the ResourceLink to be created, relative to the java:comp/env context. Default: `$name`
+The name of the ResourceLink to be created, relative to the java:comp/env context.
```
backticks around `java:comp/env`

pmcmaw commented on README.md:
```diff
@@ -767,169 +1206,248 @@ The description is an an optional string for a human-readable description of thi
+
+Optional
 
 #### tomcat::install
```
Add a space in between this title and the next sentence, this needs to be continued from here to the end of the doc.

pmcmaw replied:
Also add bacticks around `tomcat::install`

pmcmaw commented on README.md:
```diff
@@ -767,169 +1206,248 @@ The description is an an optional string for a human-readable description of thi
 
 ##### `environment`
 Environment variables for settings such as http_proxy, https_proxy, or ftp_proxy. These are passed through to the staging
 module and then to the underlying exec(s), so it follows the same format of the exec type `environment`
```
Remove hard return.

pmcmaw commented on README.md:
```diff
@@ -998,73 +1554,129 @@ The user of the jsvc process when `use_init => true`
+
+Default value: `true`.
 
 ####tomcat::war
```
add backticks around `tomcat::war`

pmcmaw commented on README.md:
```diff
@@ -998,73 +1554,129 @@ The user of the jsvc process when `use_init => true`
+Valid options: a string containing a `puppet://`, `http(s)://`, or `ftp://` URL.
 
 ##Limitations
 
```
Does \*nix need wrapped in backticks?

jbondpdx replied:
Hmm, that's tricky because it's not a code or string. Really, it's jargon, so let's replace it with Unix-like systems. (I'm guilty of using this, so I'm glad you mentioned it @pmcmaw !)

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,900 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
-Specifies the type of element to manage. Valid options: 'user' or 'role'. Default: 'user'.
+Specifies the type of element to manage.
+
+Valid options: 'user' or 'role'.
```
These are OK; strings should be in single quotes and these are strings.

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,900 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
 #####`element_name`
 
-Sets the element's username (or rolename, if `element` is set to 'role'). Valid options: a string. Default: $name.
+Sets the element's username (or rolename, if `element` is set to 'role').
```
Single quotes are fine here.

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,900 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
+
+Valid options: a string containing a Java class name.
+
+Default value: the 'name' passed in your define.
```
Single quotes OK here.

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,900 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
 #####`manage_group`
 
-Determines whether defines should default to creating the specified group, if it doesn't exist. Uses Puppet's native [`group` resource type](https://docs.puppetlabs.com/references/latest/type.html#group) with default parameters. Valid options: 'true' and 'false'. Default: 'true'.
+Determines whether defines should default to creating the specified group, if it doesn't exist. Uses Puppet's native [`group` resource type](https://docs.puppetlabs.com/references/latest/type.html#group) with default parameters.
```
Define(s) should be defined type(s). I normally wouldn't worry about it, but I'm concerned it won't be translated correctly. Apply this change throughout.

jbondpdx replied:
I'm still seeing defines (the noun, that is) instead of defined types. I think I failed at Github review before, so I'm not sure whether you saw my earlier comment. (Or if maybe I'm reviewing an old version of this doc now!)

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,900 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
+
+Valid options: `true` and `false`.
+
+Default value: `false`.  If two realms are defined for a specific server config only use purge_realms for the first realm and ensure the realms enforce a strict order between each other.
```
backticks for purge_realms (setting)

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,900 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
+Default value: `tomcat`.
 
 #### tomcat::config::properties::property
 Specifies an additional entry for the catalina.properties file a given catalina base.
```
needs a space in between, and catalina.properties should be in backticks (filename).

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,900 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
 #####`shutdown`
 
-Designates a command that shuts down Tomcat when the command is received through the specified address and port. Maps to the [shutdown XML attribute](http://tomcat.apache.org/tomcat-8.0-doc/config/server.html#Common_Attributes) Valid options: a string. Default: undef.
+Designates a command that shuts down Tomcat when the command is received through the specified address and port. Maps to the [shutdown XML attribute](http://tomcat.apache.org/tomcat-8.0-doc/config/server.html#Common_Attributes)
```
should have a period at the end of this sentence.

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,900 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
+
+Valid options: a string containing the name attribute of the Service.
+
+Default value: `Catalina`.
```
I think Catalina here is a string and should be in single quotes.

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,900 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
+
+Valid options: a string containing an absolute path.
+
+Default value: `$::tomcat/catalina_home`.
```
should be in single quotes (it's a string).

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,900 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
-Specifies whether the [Context XML element](http://tomcat.apache.org/tomcat-8.0-doc/config/context.html) should exist in the configuration file. Valid options: 'true', 'false', 'present', and 'absent'. Default: 'present'.
+Specifies whether the [Context XML element](http://tomcat.apache.org/tomcat-8.0-doc/config/context.html) should exist in the configuration file.
+
+Valid options: `true`, `false`, `present`, and `absent`.
```
true and false are correct in backticks, but present and absent are strings, rather than true Booleans, so should be in single quotes.

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,900 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
+
+Valid options: a string containing the name attribute of the Service.
+
+Default value: `Catalina`.
```
single quote, string

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,900 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
+
+Valid options: a string containing an absolute path.
+
+Default value: `${catalina_base}/config/server.xml`.
```
string, single quotes

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,900 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
-Specifies whether the [backgroundProcessorDelay XML attribute](http://tomcat.apache.org/tomcat-8.0-doc/config/engine.html#Common_Attributes) should exist in the configuration file. Valid options: 'true', 'false', 'present', and 'absent'. Default: 'present'.
+Specifies whether the [backgroundProcessorDelay XML attribute](http://tomcat.apache.org/tomcat-8.0-doc/config/engine.html#Common_Attributes) should exist in the configuration file.
+
+Valid options: `true`, `false`, `present`, and `absent`.
```
present and absent should be single quotes

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,900 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
-Specifies whether the [className XML attribute](http://tomcat.apache.org/tomcat-8.0-doc/config/engine.html#Common_Attributes) should exist in the configuration file. Valid options: 'true', 'false', 'present', and 'absent'. Default: 'present'.
+Specifies whether the [className XML attribute](http://tomcat.apache.org/tomcat-8.0-doc/config/engine.html#Common_Attributes) should exist in the configuration file.
+
+Valid options: `true`, `false`, `present`, and `absent`.
```
present and absent should be in single quotes

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,900 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
+
+Valid options: `true`, `false`, `present`, and `absent`.
+
+Default value: `present`.
```
single quotes here too

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,900 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
-Specifies whether the [jvmRoute XML attribute](http://tomcat.apache.org/tomcat-8.0-doc/config/engine.html#Common_Attributes) should exist in the configuration file. Valid options: 'true', 'false', 'present', and 'absent'. Default: 'present'.
+Specifies whether the [jvmRoute XML attribute](http://tomcat.apache.org/tomcat-8.0-doc/config/engine.html#Common_Attributes) should exist in the configuration file.
+
+Valid options: `true`, `false`, `present`, and `absent`.
```
present and absent should be single quoted as strings

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,900 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
+
+Valid options: `true`, `false`, `present`, and `absent`.
+
+Default value: `present`.
```
string, single quote

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,900 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
+
+Valid options: a string containing the name attribute of the Service.
+
+Default value: `Catalina`.
```
string, single quote

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,900 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
+
+Valid options: a string containing an absolute path.
+
+Default value: `${catalina_base}/config/server.xml`.
```
string, single quote

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,900 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
-Specifies whether the [startStopThreads XML attribute](http://tomcat.apache.org/tomcat-8.0-doc/config/engine.html#Common_Attributes) should exist in the configuration file. Valid options: 'true', 'false', 'present', and 'absent'. Default: 'present'.
+Specifies whether the [startStopThreads XML attribute](http://tomcat.apache.org/tomcat-8.0-doc/config/engine.html#Common_Attributes) should exist in the configuration file.
+
+Valid options: `true`, `false`, `present`, and `absent`.
```
present and absent strings, single quote; I'm not going to tag future instances though :-)

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,900 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
 #### tomcat::config::server::globalnamingresource
 
-Configure GlobalNamingResources Resource elements in $CATALINA\_BASE/conf/server.xml
+Configure GlobalNamingResources Resource elements in `$CATALINA_BASE/conf/server.xml`
```
for the record, this is perfect: the location should be in backticks unless it's a string value the user is supposed to input.

I know, it's kind of weird and confusing. We're working on that. :)

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,900 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
-_Note: This is used verbatim in your configuration so make sure the case is correct._
+Specifies the type of element to create
+
+Valid options: . `Resource`, `Environment` or any other valid node.
```
I am not sure if the '.' is a value or a typo. These values are strings (even though we don't say that), so should be in single quotes.

jbondpdx commented on README.md:
```diff
@@ -767,169 +1206,248 @@ The description is an an optional string for a human-readable description of thi
-Specifies the owner of the source installation directory. Default: `$::tomcat::user`
+Specifies the owner of the source installation directory.
+
+Default value: `$::tomcat::user`
```
this (and the other default values like it) are strings, so should be in single quotes

jbondpdx commented on README.md:
```diff
@@ -767,169 +1206,248 @@ The description is an an optional string for a human-readable description of thi
 
 ##### `manage_home`
-Specifies whether the directory of catalina\_home should be managed by puppet. This may not be preferable in network filesystem environments. Default: `true`
+Specifies whether the directory of catalina_home should be managed by puppet. This may not be preferable in network filesystem environments.
```
Puppet should be capitalized

jbondpdx commented on README.md:
```diff
@@ -767,169 +1206,248 @@ The description is an an optional string for a human-readable description of thi
-Determines whether the specified package should be installed. Only valid if `install_from_source` is set to `false`. Maps to the `ensure` parameter of Puppet's native [`package` resource type](https://docs.puppetlabs.com/references/latest/type.html#package). Default: 'present'.
+Determines whether the specified package should be installed. Only valid if `install_from_source` is set to `false`. Maps to the `ensure` parameter of Puppet's native [`package` resource type](https://docs.puppetlabs.com/references/latest/type.html#package).
+
+Default value: `present`.
```
string, single quote

jbondpdx commented on README.md:
```diff
@@ -767,169 +1206,248 @@ The description is an an optional string for a human-readable description of thi
 
 ##### `manage_base`
-Specifies whether the directory of catalina\_base should be managed by puppet. This may not be preferable in network filesystem environments. Default: `true`
+Specifies whether the directory of catalina_base should be managed by puppet. This may not be preferable in network filesystem environments.
```
capitalize Puppet

jbondpdx commented on README.md:
```diff
@@ -767,169 +1206,248 @@ The description is an an optional string for a human-readable description of thi
-Valid options: a string containing an absolute path. Default: `$catalina_home`
+Valid options: a string containing an absolute path.
+
+Default value: `$catalina_home`
```
These "Default value" lines should all have periods at the end, after the backtick or single quote.

wilson208 commented:
@jbondpdx @pmcmaw have added this commit which should address all your comments:
https://github.com/puppetlabs/puppetlabs-tomcat/pull/202/commits/24a3bf084892293577dabca56145707a45b8b8cf

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,901 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
+
+Valid options: a string containing an absolute path.
+
+Default value: `/opt/apache-tomcat`.
```
String values go in single quotes

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,901 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
+
+Valid options: a string containing a valid group name.
+
+Default value: `tomcat`.
```
String values go in single quotes

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,901 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
+
+Valid options: a string containing a valid username.
+
+Default value: `tomcat`.
```
String values go in single quotes

jbondpdx commented on README.md:
```diff
@@ -236,479 +236,901 @@ The base class sets defaults used by other defined types, such as `tomcat::insta
+
+Valid options: a string containing an absolute path.
+
+Default value: `$::tomcat::catalina_home`.
```
String values go in single quotes

jbondpdx commented on README.md:
```diff
@@ -723,268 +1145,451 @@ Specifies a Manager element in the designated xml configuration.
-Specifies the root of the Tomcat installation. Default: `$tomcat::catalina_home`
+Specifies the root of the Tomcat installation.
+
+Default value: `$tomcat::catalina_home`.
```
String values go in single quotes

jbondpdx commented on README.md:
```diff
@@ -723,268 +1145,451 @@ Specifies a Manager element in the designated xml configuration.
-Specifies the root of the Tomcat installation. Default: `$tomcat::catalina_home`
+Specifies the root of the Tomcat installation.
+
+Default value: `$tomcat::catalina_home`.
```
String values go in single quotes

jbondpdx commented on README.md:
```diff
@@ -723,268 +1145,451 @@ Specifies a Manager element in the designated xml configuration.
+
+Valid options: a string containing an absolute path.
+
+Default value: `$::tomcat::catalina_home`.
```
String values go in single quotes

jbondpdx commented on README.md:
```diff
@@ -723,268 +1145,451 @@ Specifies a Manager element in the designated xml configuration.
-Valid options: a string containing an absolute path. Default: `$catalina_home`
+Valid options: a string containing an absolute path.
+
+Default value: `$catalina_home`.
```
String values go in single quotes

jbondpdx commented on README.md:
```diff
@@ -723,268 +1145,451 @@ Specifies a Manager element in the designated xml configuration.
-Valid options: a string containing an absolute path. Default: `$::tomcat::catalina_home`
+Valid options: a string containing an absolute path.
+
+Default value: `$::tomcat::catalina_home`.
```
String values go in single quotes

jbondpdx commented on README.md:
```diff
@@ -723,268 +1145,451 @@ Specifies a Manager element in the designated xml configuration.
+
+Valid options: a string containing an absolute path.
+
+Default value: `$::tomcat::catalina_home`.
```
String values go in single quotes

jbondpdx commented on README.md:
```diff
@@ -723,268 +1145,451 @@ Specifies a Manager element in the designated xml configuration.
+
+Valid options: a string containing an absolute path.
+
+Default value: `$::tomcat::catalina_home`.
```
String values go in single quotes

wilson208 commented:
Hey @jbondpdx, thanks a lot for reviewing this for me. Sorry I missed a few things last time, I have updated this PR with the last of your comments.

The commit messages aren't very descriptive on the last 2 commits, I didnt rebase them so you can see what I changed in each iteration. Let me know when you are happy with this and I will rebase and squash these in to one.

Thanks!

jbondpdx commented:
Thank you again, @wilson208 !
